network/src/receiver.rs

```diff
@@ -87,4 +96,23 @@ impl<Handler: MessageHandler> Receiver<Handler> {
             warn!("Connection closed by peer {peer}");
         });
     }
+
+    /// Spawns a new task that will listen for any responses from the accounting layer and
+    /// forward them to the tcp writer.
+    async fn spawn_response_writer(
+        mut tr_response: mspcReceiver<Bytes>,
+        mut tcp_writer: TcpWriter,
+    ) {
+        tokio::spawn(async move {
+            loop {
+                tokio::select! {
+                   Some(response) = tr_response.recv() => {
```

Comment:
This branch has a plausible story of being disabled (because the future is not ready, with the handler not having fired a response) at many passes through the loop. Because this `select!` call has no `else`, this would trigger a panic. Let's avoid it.

Reply:
Thanks for this comment, what do you mean by unwind our response task?
I assumed that mspc channels behave as oneshot channels and this loop would stop when the channel was closed or dropped.